teku/src/main/java/tech/pegasys/teku/cli/options/DepositOptions.java

```diff
@@ -68,17 +69,35 @@ public class DepositOptions {
       fallbackValue = "true")
   private boolean depositSnapshotEnabled = PowchainConfiguration.DEFAULT_DEPOSIT_SNAPSHOT_ENABLED;
 
+  @CommandLine.Spec CommandLine.Model.CommandSpec commandSpec;
+
   public void configure(final TekuConfiguration.Builder builder) {
-    builder.powchain(
-        b ->
-            b.eth1Endpoints(eth1Endpoints)
-                .eth1LogsMaxBlockRange(eth1LogsMaxBlockRange)
-                .useMissingDepositEventLogging(useMissingDepositEventLogging)
-                .depositSnapshotPath(depositSnapshotPath)
-                .depositSnapshotEnabled(parseDepositSnapshotEnabled()));
-  }
+    final CommandLine.ParseResult parseResult = commandSpec.commandLine().getParseResult();
+
+    final String checkpointSyncUrlCliOption = "--checkpoint-sync-url";
+    final boolean checkpointSyncUrlSet = parseResult.hasMatchedOption(checkpointSyncUrlCliOption);
```

Comment:
I don't like this way of using data from other options module. Is it possible to implement the same logic more casual way by checking options here and there in `TekuConfiguration.java`? As initialState could be set not only by "--checkpoint-sync-url" we could add the property checkpointSyncUrlSet as boolean to `Eth2NetworkConfiguration to clarify this case.